c++/src/kj/compat/http.c++

```diff
@@ -5674,11 +5679,18 @@ public:
     return getClient(parsed).openWebSocket(path, headersCopy);
   }
 
-  ConnectRequest connect(kj::StringPtr host, const HttpHeaders& headers) override {
+  ConnectRequest connect(
+      kj::StringPtr host, const HttpHeaders& headers, ConnectSettings settings) override {
     // We want to connect directly instead of going through a proxy here.
     // https://github.com/capnproto/capnproto/pull/1454#discussion_r900414879
+    kj::Maybe<kj::Promise<kj::Own<kj::NetworkAddress>>> addr;
+    if (settings.useTls) {
+      addr = KJ_ASSERT_NONNULL(tlsNetwork).parseAddress(host);
```

Comment:
This is an error on the caller's part, so should be REQUIRE instead of ASSERT.

```suggestion
      addr = KJ_REQUIRE_NONNULL(tlsNetwork, "this HttpClient doesn't support TLS")
          .parseAddress(host);
```